packages/rollup-plugin/src/index.ts

```diff
@@ -1,14 +1,51 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import {
-  codecovUnpluginFactory,
+  type UnpluginOptions,
+  createRollupPlugin,
+  type RollupPlugin,
+} from "unplugin";
+import {
+  normalizeOptions,
+  red,
   type Options,
+  checkNodeVersion,
 } from "@codecov/bundler-plugin-core";
 
 import { rollupBundleAnalysisPlugin } from "./rollup-bundle-analysis/rollupBundleAnalysisPlugin";
 
-const codecovUnplugin = codecovUnpluginFactory({
-  bundleAnalysisUploadPlugin: rollupBundleAnalysisPlugin,
-});
+const codecovRollupPluginFactory = createRollupPlugin<Options, true>(
+  (userOptions, unpluginMetaContext) => {
+    const plugins: UnpluginOptions[] = [];
+
+    const normalizedOptions = normalizeOptions(userOptions);
+
+    if (!normalizedOptions.success) {
+      for (const error of normalizedOptions.errors) {
+        red(error);
+      }
+      return [];
+    }
+
+    if (checkNodeVersion(unpluginMetaContext)) {
+      return [];
+    }
+
+    const options = normalizedOptions.options;
+    if (options?.enableBundleAnalysis) {
```

Comment:
Why is `options` undefined here after we've normalized it? This should be `options?.enableBundleAnalysis` right?

Reply:
Just a force of habit from writing so much plain JS ...